### PR TITLE
feat: container resource banner depends on selected connection

### PR DESCRIPTION
**Pull request by axel7083**
### What does this PR do?

Makes the resource warning depends on the connection selected. 

### Screenshot / video of UI

https://github.com/user-attachments/assets/b048fd19-015b-4ece-8423-d0977f433989

### What issues does this PR fix or reference?

Fixes https://github.com/containers/podman-desktop-extension-ai-lab/issues/1611

### How to test this PR?

- [x] unit tests has been provided

**Manually**

- Create a podman machine with little memory (not possible on windows) and one with a lot of memory. 
> `podman machine init --memory=1024 machine-low-memory --now` 
- assert both machines are running
- Go to AI Lab 
- Navigate to `Creating Model service`
- assert both machine are visible
- Select the low memory machine
- assert warning banner is visible
- Select the high memory machine
- assert warning banner is **not** visible

### CODE REVIEW

File: packages/shared/src/models/IContainerConnectionInfo.ts [hunk 1]
```diff
@@ -27,11 +27,12 @@ export interface ContainerProviderConnectionInfo {
...
 
 export interface CheckContainerConnectionResourcesOptions {
-  modelInfo: ModelCheckerInfo;
+  model: ModelInfo & { memory: number };
+  context: ModelCheckerContext;
   connection?: ContainerProviderConnectionInfo;
 }
 
-export type ContainerConnectionInfo =
+export type ContainerConnectionResourceInfo =
```
Comment (lstocchi) [inline packages/shared/src/models/IContainerConnectionInfo.ts:35]: i don't get why it was renamed as only one of the subtypes directly refers to resources.  So in most cases there are no info about resources

**Review by lstocchi — APPROVED** [review]
LGTM, only doubts about the renaming.

**Review by jeffmaury — APPROVED** [review]
LGTM

But the included refactoring (renaming) make it harder to review and should be separate

File: packages/shared/src/models/IContainerConnectionInfo.ts (hunk 1 above)
Comment (axel7083) [reply packages/shared/src/models/IContainerConnectionInfo.ts]: I revert the rename

**Pull request merged**